### PR TITLE
Add Configurable pooling variants for the Deep Sets model

### PR DESCRIPTION
**Pull request by Nedak23**
Addresses #122

- Makes DeepSetsClassifier pooling configurable via a pooling config key, supporting: mean, max, sum, mean_max, mean_max_logcount (default)
- Adds the option to DEFAULT_CONFIG and configs/deepsets_ispy2.yaml with backward-compatible defaults
- Fixes max-pooling to use a portable loop (compatible with PyTorch <1.12)
Adds 22 parametrized tests in tests/test_deepsets_model.py

### CODE REVIEW

**Review by jpivarski — APPROVED** [review]
This addresses some points in #122, but not all of them, so I removed it from the "would close" list. The issue also asks for:
  - a short summary of the current baseline setup,
  - fair comparison of pooling variants on the same dataset/folds,
  - one compact result table,
  - selection of a default pooling based on that comparison.

But this PR does its focused task: it implements configurable pooling in the model/training path, and it doesn't seem to have any bugs. The test coverage checks shapes, gradients, defaults, and invalid names, but it doesn't check to see if the pooling is correctly implemented with any numerical tests.

It's in a good state to merge before moving on to the next PR.
